app/models/product.rb

```diff
@@ -1,3 +1,10 @@
 class Product < ApplicationRecord
-
+   validates :name, :price, :net_price, presence: true
+   validates :new_product, inclusion: { in: [true, false] }, presence: true
+   validates :catalogue_number, presence: true, uniqueness: true, on: :create and :update
```

Comment:
Skoro i update i create to można to poprostu wywalić. Wtedy walidacja działa przy tworzeniu nowego i edycji istniejącego obiektu